pmd-plsql/etc/grammar/PLSQL.jjt

```diff
@@ -2721,23 +2721,16 @@ ASTMergeStatement MergeStatement() :
         |   [ LOOKAHEAD(2) SchemaName() "." ] TableName() [ TableAlias() ]
     )
     <ON> "(" Condition() ")"
-    [ LOOKAHEAD(MergeUpdateClausePrefix()) MergeUpdateClause() ]
-    [ LOOKAHEAD(2) MergeInsertClause() ]
+    ( <WHEN> ( ( KEYWORD("MATCHED") MergeUpdateClause() )  | ( <NOT> KEYWORD("MATCHED") MergeInsertClause() ) ) ) +
     [ ErrorLoggingClause() ]
     [ ReturningClause() ]
     { return jjtThis; }
 }
 
-void MergeUpdateClausePrefix() #void:
-{}
-{
-    <WHEN> KEYWORD("MATCHED")
-}
-
```

Comment:
That's the method you removed. You'll need to re-add this...

Reply:
I removed it because I don't need it (and actualy spend a lot of time getting it to work with my change and failed).
Now from the error I understand that this is a breaking change.  But adding it without it actually being used somewhere would make the tool happy, but that is just as breaking if some code would depend upon it being called.  The later would however be more difficult to find as the function is still there.

But I have no problem with adding it again. If possible with some Deprecated indication.

Reply:
We forgot to make this generated class package private, so it lives now as public API, but the whole class is already deprecated (since 7.3.0).

You can just leave the method there and don't use it. That would suffice our (weak) compatibility requirements.